To: vim_dev@googlegroups.com
Subject: Patch 7.3.042
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.042
Problem:    No spell highlighting when re-using an empty buffer.
Solution:   Clear the spell checking info only when clearing the options for a
	    buffer. (James Vega)
Files:	    src/buffer.c


*** ../vim-7.3.041/src/buffer.c	2010-09-21 16:56:29.000000000 +0200
--- src/buffer.c	2010-10-27 16:09:26.000000000 +0200
***************
*** 639,644 ****
--- 639,647 ----
      {
  	clear_wininfo(buf);		/* including window-local options */
  	free_buf_options(buf, TRUE);
+ #ifdef FEAT_SPELL
+ 	ga_clear(&buf->b_s.b_langp);
+ #endif
      }
  #ifdef FEAT_EVAL
      vars_clear(&buf->b_vars.dv_hashtab); /* free all internal variables */
***************
*** 661,669 ****
      vim_free(buf->b_start_fenc);
      buf->b_start_fenc = NULL;
  #endif
- #ifdef FEAT_SPELL
-     ga_clear(&buf->b_s.b_langp);
- #endif
  }
  
  /*
--- 664,669 ----
*** ../vim-7.3.041/src/version.c	2010-10-27 16:01:17.000000000 +0200
--- src/version.c	2010-10-27 16:10:33.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     42,
  /**/

-- 
BLACK KNIGHT:  I move for no man.
ARTHUR:        So be it!
    [hah] [parry thrust]
    [ARTHUR chops the BLACK KNIGHT's left arm off]
ARTHUR:        Now stand aside, worthy adversary.
BLACK KNIGHT:  'Tis but a scratch.
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
